### PR TITLE
Validate birthday in austrian insurance number

### PR DESCRIPTION
**Pull request by lkrempler**

### CODE REVIEW

File: src/Rules/AustrianInsuranceNumber.php [hunk 1]
```diff
@@ -51,6 +52,25 @@ private function startsNotWithZero(string $svnumber): bool
         return (int) $svnumber[0] !== 0;
     }
 
+    private function hasValidBirthday(string $svnumber): bool
+    {
+        $splittedBirthday = str_split(substr($svnumber, 4, 10), 2);
```
Comment (olivervogel) [inline src/Rules/AustrianInsuranceNumber.php:57]: From the [specification](https://www.sozialversicherung.at/cdscontent/?contentid=10007.820902&viewmode=content#datareplacementdoneWorausbestehtdieVersicherungsnummer820902) I can see that the date of birth is in the 5-10 digit. The `substr` call, on the other hand, fetches the digits 5-15 `$offset` to `$offset` + `$length`. This is irrelevant for the 10-digit number, as it is always read to the end. This means that the length can also be omitted directly.

```suggestion
        $splittedBirthday = str_split(substr($svnumber, 4), 2);
```

File: src/Rules/AustrianInsuranceNumber.php [hunk 2]
```diff
@@ -38,6 +38,7 @@ public function isValid(mixed $value): bool
         return is_numeric($value)
             && $this->startsNotWithZero($value)
             && $this->hasValidLength($value)
+            && $this->hasValidBirthday($value)
```
Comment (olivervogel) [inline src/Rules/AustrianInsuranceNumber.php:41]: I read in the [spec](https://www.sozialversicherung.at/cdscontent/?contentid=10007.820902&viewmode=content#datareplacementdoneWorausbestehtdieVersicherungsnummer820902) that the date of birth does not always appear in the number. Does this mean that these particular numbers will no longer be recognized as valid after the patch?

**Review by olivervogel — COMMENTED** [review]
Thanks for improving.

File: src/Rules/AustrianInsuranceNumber.php (hunk 1 above)
Comment (lkrempler) [reply src/Rules/AustrianInsuranceNumber.php]: Thanks!

File: src/Rules/AustrianInsuranceNumber.php (hunk 2 above)
Comment (lkrempler) [reply src/Rules/AustrianInsuranceNumber.php:41]: The date of birth must always be stated. If it is not known exactly, the day is always the 1st, the month may be greater than 12.
If the date of birth is missing, the number is invalid.

**Pull request merged**